container.go

```diff
@@ -750,3 +774,45 @@ func (c *Container) detachDevices() error {
 
 	return nil
 }
+
+func (c *Container) addResources() error {
+	//TODO add support for memory, Issue: https://github.com/containers/virtcontainers/issues/578
+	if c.config != nil {
+		cType := ContainerType(c.config.Annotations[vcAnnotations.ContainerTypeKey])
+		if cType == PodSandbox {
+			virtLog.Debugf("Cannot hot add vCPUs to sandbox containers")
+			return nil
+		}
+
+		vCPUs := c.config.Resources.VCPUs
+		if vCPUs != 0 {
+			virtLog.Debugf("hot adding %d vCPUs", vCPUs)
+			if err := c.pod.hypervisor.hotplugAddDevice(uint32(vCPUs), cpuDev); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (c *Container) removeResources() error {
+	//TODO add support for memory, Issue: https://github.com/containers/virtcontainers/issues/578
+	if c.config != nil {
+		cType := ContainerType(c.config.Annotations[vcAnnotations.ContainerTypeKey])
+		if cType == PodSandbox {
+			virtLog.Debugf("Cannot hot remove vCPUs from sandbox containers")
+			return nil
+		}
+
+		vCPUs := c.config.Resources.VCPUs
```

Comment:
Same comment, we should be able to rely on cpu quota and period I think.

Reply:
the problem with that is if the formula changes, let's suppose you start a container with a constrain of 4.5 CPUs, currently 5 vCPUs are added, but if the formula changes in the future to return 6 vCPUs and you update the runtime, when that container ends, the runtime will try to hot-remove 6 vCPUs, not 5. From my point of view the resources assigned to the containers should be saved to avoid this kind of issues.

Reply:
If the formula changes (thing very odd IMO), then we'll change it. But it's easier to get pods being started with a default number of CPUs (unless otherwise noticed by configuration.toml), and add hotplug CPUs for new containers. I would like the Docker case to follow the same logic.